src/main.rs

```diff
@@ -33,4 +34,12 @@ fn main() {
     };
 
     engine.start_application(&sample_app);
+
+    let _ = match server::start_server() {
+        Ok(server) => server,
+        Err(e) => {
+            info!("Failed to start gRPC server: {}", e);
+            return;
+        }
+    };
```

Comment:
If you don't use the return value, can you return nothing on success and remove the assignation?

It would probably also be better to propagate the error and use a more meaningful log level.
```suggestion
    match server::start_server() {
        Ok(_) => Ok(()),
        Err(e) => {
            error!("Failed to start gRPC server: {}", e);
            Err(e)
        }
    }
```